xml/cap_depl_eks.xml

```diff
@@ -370,151 +288,347 @@ Custom TCP Rule    TCP          4443            0.0.0.0/0       CAP WSS
+   <para>
+    Some UAA and SCF pods perform only deployment tasks, and it is normal for
+    them to show as unready after they have completed their tasks, as these
+    examples show:
+   </para>
+<screen>
+&prompt.user;kubectl get pods --namespace uaa
+secret-generation-1-z4nlz   0/1       Completed
+          
+&prompt.user;kubectl get pods --namespace scf
+secret-generation-1-m6k2h       0/1       Completed
+post-deployment-setup-1-hnpln   0/1       Completed</screen>
+  </important>
+ </sect1>
+ 
+ <sect1 xml:id="sec.cap.install-scf-eks">
+  <title>Deploy SCF</title>
+    <para>
+      Because of the way EKS runs health checks, &cap; requires an edit to 
+      one of the <literal>scf</literal> &helm; charts, and 
+      then after a successful <literal>scf</literal> deployment remove a 
+      listening port from the EKS load balancer's listeners list.  
+  </para>
+  <para>
+      On your remote admin machine, look in
+      <filename>~.helm/cache/archive/cf-2.15.2.tgz</filename>. Extract the
+      <filename>cf-2.15.2.tgz</filename>archive, and add this to the 
+      <literal>ports</literal> section of
+      <filename>templates/tcp-router.yaml</filename>:
+  </para>
+  <screen>
+- name: "healthcheck"
+  port: 8080
+  protocol: "TCP"
+  targetPort: "healthcheck"
+</screen>
+<para>
+    The section should look like this:
+</para>
+<screen>
+    ports:
+    - name: "healthcheck"
+      port: 8080
+      protocol: "TCP"
+      targetPort: "healthcheck"
+    {{- range $port := until (int .Values.sizing.tcp_router.ports.tcp_route.count) }}
+    - name: "tcp-route-{{ $port }}"
+      port: {{ add 20000 $port }}
+      protocol: "TCP"
+      targetPort: "tcp-route-{{ $port }}"
+    {{- end }}
+    selector:
+      app.kubernetes.io/component: "tcp-router" 
+      </screen>
+      
+  <para>
+   Now pass your UAA secret and certificate to SCF, then use &helm; to
+   install &suse; &cf;:
+  </para>
+
+<screen>&prompt.user;SECRET=$(kubectl get pods --namespace uaa \
+-o jsonpath='{.items[?(.metadata.name=="uaa-0")].spec.containers[?(.name=="uaa")].env[?(.name=="INTERNAL_CA_CERT")].valueFrom.secretKeyRef.name}')
+
+&prompt.user;CA_CERT="$(kubectl get secret $SECRET --namespace uaa \
+-o jsonpath="{.data['internal-ca-cert']}" | base64 --decode -)"
+
+&prompt.user;helm install suse/cf \
+--name <replaceable>susecf-scf</replaceable> \
+--namespace scf \
+--values scf-config-values.yaml \
+--set "secrets.UAA_CA_CERT=${CA_CERT}"</screen>
+
+  <para>
+   Sit back and wait for the pods come online:
+  </para>
+
+<screen>&prompt.user;watch -c 'kubectl get pods --namespace scf'</screen>
+
+  <para>
+   When all services are running, verify that the load balancer has been created
+   by viewing it in your EC2 Dashboard, or by fetching its hostname: 
+  </para>
+  <screen>
+&prompt.user;kubectl get service -o jsonpath='{.status.loadBalancer.ingress[0].hostname}'
+a3872f427e6s490cb0a14c38ad8e-9362763047.us-west-2.elb.amazonaws.com
```

Comment:
I didn't check if this is the actual thing… we probably want to replace the `936…` part with something random that's not actually us (but then I'm paranoid).

Reply:
:heavy_check_mark:

Reply:
:heavy_check_mark: